File: Task2/Task2.java
import java.util.Scanner;

public class Main {
    public static int counter(int[] nums) {
        int count = 0;
        for (int num : nums) {
            int digits = (num == 0) ? 1 : 0;
            int n = Math.abs(num);
            while (n != 0) {
                n /= 10;
                digits++;
            }
            if (digits % 2 != 0) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.print("Введите числа через запятую: ");
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] strNums = input.split(",");
        int[] nums = new int[strNums.length];
        for (int i = 0; i < strNums.length; i++) {
            nums[i] = Integer.parseInt(strNums[i].trim());
        }
        System.out.println(counter(nums));
    }
}

File: Task2/Task2.cs
using System;
using System.Linq;

class Program
{
    static int counter(int[] nums)
    {
        int count = 0;
        foreach (var number in nums)
        {
            int digits = (number == 0) ? 1 : 0;
            int num = Math.Abs(number);

            while (num != 0)
            {
                num /= 10;
                digits++;
            }
            if (digits % 2 != 0)
            {
                count++;
            }
        }
        return count;
    }

    static void Main()
    {
        Console.WriteLine("Введите числа через запятую: ");
        var input = Console.ReadLine();
        var nums = input.Split(',').Select(int.Parse).ToArray();
        Console.WriteLine(counter(nums));
    }
}